### PR TITLE
Fix brace completion for the with expression

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

**Comment by agocke** [comment]
fyi @jcouv

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs [hunk 1]
```diff
@@ -208,6 +208,10 @@ public static (SyntaxToken openBrace, SyntaxToken closeBrace) GetBraces(this Syn
                     return (switchExpression.OpenBraceToken, switchExpression.CloseBraceToken);
                 case PropertyPatternClauseSyntax property:
                     return (property.OpenBraceToken, property.CloseBraceToken);
+#if !CODE_STYLE
+                case WithExpressionSyntax withExpr:
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs:212]:
```suggestion
                case SyntaxNode withExpr when (withExpr.IsKind(SyntaxKindEx.WithExpression)):
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs [hunk 2]
```diff
@@ -208,6 +208,10 @@ public static (SyntaxToken openBrace, SyntaxToken closeBrace) GetBraces(this Syn
...
+                    return (withExpr.OpenBraceToken, withExpr.CloseBraceToken);
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs:213]: ❔ Are `OpenBraceToken` and `CloseBraceToken` the first and last tokens of this node (respectively)?

Comment (agocke) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs:213]: Nope

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs (hunk 1 above)
Comment (agocke) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs]: Wait, this doesn't work. I need the stronger type.

**Pull request merged**